lib/features/workout/data/db/workout_entity.dart

```diff
@@ -0,0 +1,187 @@
+import 'package:flex_workout_mobile/features/exercise/data/db/exercise_entity.dart';
+import 'package:flex_workout_mobile/features/workout/data/models/workout_model.dart';
+import 'package:objectbox/objectbox.dart';
+
+@Entity()
+class WorkoutEntity {
+  WorkoutEntity({
+    required this.title,
+    required this.subtitle,
+    required this.description,
+    required this.updatedAt,
+    required this.createdAt,
+    this.id = 0,
+  });
+
+  @Backlink('workout')
+  final sections = ToMany<WorkoutSectionEntity>();
+
+  @Id()
+  int id = 0;
+
+  String title;
+  String subtitle;
+  String description;
+
+  @Property(type: PropertyType.date)
+  DateTime updatedAt;
+  @Property(type: PropertyType.date)
+  DateTime createdAt;
+}
+
+extension ConvertWorkout on WorkoutEntity {
+  WorkoutModel toModel() => WorkoutModel(
+        id: id,
+        title: title,
+        subtitle: subtitle,
+        description: description,
+        createdAt: createdAt,
+        updatedAt: updatedAt,
+        sections: sections.map((e) => e.toModel()).toList(),
+      );
+}
+
+@Entity()
+class WorkoutSectionEntity {
+  WorkoutSectionEntity({this.id = 0});
+
+  @Id()
+  int id = 0;
+
+  final workout = ToOne<WorkoutEntity>();
+
+  final defaultSection = ToOne<DefaultSectionEntity>();
+  final supersetSection = ToOne<SupersetSectionEntity>();
+}
+
+extension ConvertSection on WorkoutSectionEntity {
+  IWorkoutSection toModel() {
+    if (defaultSection.target != null) {
+      return defaultSection.target!.toModel();
+    } else if (supersetSection.target != null) {
+      return supersetSection.target!.toModel();
+    } else {
+      return defaultSection.target!.toModel();
```

Comment:
This will always throw an error, it is better to remove it entirely

Reply:
can you provide the error it will throw?

Reply:
if i remember correctly, an error was thrown before i added the super set section